src/OrchardCore/OrchardCore/OrchardCore.csproj

```diff
@@ -21,6 +21,7 @@
   <ItemGroup>
     <ProjectReference Include="..\OrchardCore.Abstractions\OrchardCore.Abstractions.csproj" />
     <ProjectReference Include="..\OrchardCore.Autoroute.Abstractions\OrchardCore.Autoroute.Abstractions.csproj" />
+    <ProjectReference Include="..\OrchardCore.Autoroute.Core\OrchardCore.Autoroute.Core.csproj" />
```

Comment:
I don't think it's a right reference, I presume that the service located in `OrchardCore.Autoroute.Abstractions` which is a right reference not the implementation one

Reply:
Actually, I used the class directly here instead of the interface. I'm not injecting this service because of the fact that `ServiceCollectionExtensions` is where I use it and it is where we register our services with the DI. I need to make a direct reference to the class to have an instance that I can use. Also, not using the abstraction project ISlugService because I want to be sure I'm using the proper implementation of that SlugService.

Reply:
Why the proper implementation is not registered by the Autoroute module as we did for all the services?

Reply:
The ISlugService is registered by the OrchardCore.Liquid module because we have a Liquid filter. If we enable the Slugify service on the Autoroute module it may cause issues if the Autoroute module is disabled. That's mainly why.

Reply:
Ya, I forgot it's registered inside `OC.Liquid`